WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderStatusSection.swift

```diff
@@ -0,0 +1,88 @@
+import SwiftUI
+import Yosemite
+
+struct OrderStatusSection: View {
+    let geometry: GeometryProxy
+    let dateCreated: Date
+    let orderStatus: OrderStatus
+
+    var body: some View {
+        Divider()
+
+        VStack(alignment: .leading) {
+            Text(dateString)
+                .footnoteStyle()
+
+            HStack {
+                Text(statusBadgeTitle)
+                    .foregroundColor(.black)
+                    .footnoteStyle()
+                    .padding(.horizontal, Layout.StatusBadge.horizontalPadding)
+                    .padding(.vertical, Layout.StatusBadge.verticalPadding)
+                    .background(Color(statusBadgeColor))
+                    .cornerRadius(Layout.StatusBadge.cornerRadius)
+                Spacer()
+                Button(Localization.editButton) {}
+                    .buttonStyle(LinkButtonStyle())
+                    .fixedSize(horizontal: true, vertical: true)
+                    .padding(.trailing, -Layout.linkButtonTrailingPadding) // remove trailing padding to align button title to the side
+            }
+        }
+        .padding(.horizontal, insets: geometry.safeAreaInsets)
+        .padding([.leading, .trailing, .top])
+        .background(Color(.listForeground))
+
+        Divider()
+    }
+
+    private var dateString: String {
+        let formatter = DateFormatter.mediumLengthLocalizedDateFormatter
+
+        return formatter.string(from: dateCreated)
+    }
+
+    private var statusBadgeTitle: String {
+        orderStatus.name ?? orderStatus.slug
+    }
+
+    private var statusBadgeColor: UIColor {
+        switch orderStatus.status {
+        case .pending, .completed, .cancelled, .refunded, .custom:
+            return .gray(.shade5)
+        case .onHold:
+            return .withColorStudio(.orange, shade: .shade5)
+        case .processing:
+            return .withColorStudio(.green, shade: .shade5)
+        case .failed:
+            return .withColorStudio(.red, shade: .shade5)
+        }
```

Comment:
Copied from [UILabel+OrderStatus](https://github.com/woocommerce/woocommerce-ios/blob/16d830db8e52a7c0f5764f6fccbc101563b95952/WooCommerce/Classes/Extensions/UILabel%2BOrderStatus.swift#L28).

Reply:
Do you think it's worth creating a modifier as we did for text styles? 
Like `.background(for: Status)` Also `Status` could be its own enum to not couple styles with a Networking model!

Reply:
I think single use doesn't justify app-wide modifier yet. But I've moved it to `StatusBadgeViewModel` away from view code.